cpp/include/cudf/io/parquet.hpp

```diff
@@ -401,6 +401,9 @@ class parquet_writer_options {
   bool _write_timestamps_as_int96 = false;
   // Column chunks file path to be set in the raw output metadata
   std::string _column_chunks_file_path;
+  /// vector of precision values for decimal writing. Exactly one entry
+  /// per decimal column.
+  std::vector<uint8_t>* _decimal_precisions = nullptr;
```

Comment:
While I agree with the encapsulation issue, I was avoiding the copy into the options and then a copy into the state. Since it could be a chunked call over multiple calls we have to store a copy in the state. I didn't want to make multiple copies of the vector as I'm unsure just how big this could be. I'm happy to make this change if I am overthinking the cost of this copy.

Reply:
IMO the cost of the copy will be negligible compared to the overall time to perform the write, given that the vector has only a byte per column. I'm more concerned about sneaky dangling pointer bugs. Maybe the best argument for the copy is that it's consistent with other options. If it turns out that the copies are expensive, we can come up with a better solution for all array options.

Reply:
Works for me. I feel better about holding a copy of the vector in here, but I was modeling this after the table metadata which is doing some pointer stuff. Honestly, that seems to point to the chunked writing coming in later and avoiding breaking the API.

Reply:
Pretty sure copying a even a 1000 byte vector is nbd. And if the number of columns exceeds 1000 then we have bigger problems.